packages/extension/src/ui/services/useOnClickOutside.ts

```diff
@@ -0,0 +1,30 @@
+import { RefObject, useEffect, useRef } from "react"
+
+export default function useOnClickOutside<T extends HTMLElement>(
+  node: RefObject<T | undefined>,
```

Comment:
don't use generics when not needed

```suggestion
export default function useOnClickOutside(
  node: RefObject<HTMLElement | undefined>,
```

Reply:
I think here generics are required so that we can use any HTMLElement, such as HTMLDivElement that we are using. This is a reusable hook, therefore generics are required.

Also,
<img width="1289" alt="Screenshot 2022-05-30 at 3 31 39 PM" src="https://user-images.githubusercontent.com/50968441/170969181-692d1a1d-d19a-43a5-a55e-b46104cd1a06.png">

Reply:
they are not required, you need to change the `useOnClickOutside<HTMLElement>` to `useOnClickOutside` like I suggested